packages/datetime/src/dateRangeInput.tsx

```diff
@@ -362,6 +331,30 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         this.setState(nextState);
     }
 
+    private renderInputGroup = (boundary: DateRangeBoundary) => {
+        const inputProps = boundary === DateRangeBoundary.START ? this.props.startInputProps : this.props.endInputProps;
+        const htmlProps = this.getHtmlPropsFromInputProps(inputProps);
+
+        const handleInputEvent = (e: React.SyntheticEvent<HTMLInputElement>) => this.handleInputEvent(e, boundary);
```

Comment:
Introduced `this.handleStartInputEvent` and `this.handleEndInputEvent`, plus a ternary to pick one.